app/models/manager_refresh/inventory_collection/builder.rb

```diff
@@ -0,0 +1,248 @@
+        )
+      end
+
+      protected
+
+      # Extends source hash with
+      # - a) all keys from dest (overwrite mode)
+      # - b) missing keys (missing mode)
+      #
+      # @param mode [Symbol] :overwrite | :if_missing
+      def merge_hashes(source, dest, mode)
+        return source if source.nil? || dest.nil?
+
+        if mode == :overwrite
+          source.merge(dest)
+        else
+          dest.merge(source)
+        end
+      end
+
+      # Derives model_class from persister class and @name
+      # 1) searches for class in provider
+      # 2) if not found, searches class in core
+      # Can be disabled by options :auto_model_class => false
+      #
+      # @example derives model_class from amazon
+      #
+      #   @persister_class = ManageIQ::Providers::Amazon::Inventory::Persister::CloudManager
+      #   @name = :vms
+      #
+      #   returns - <provider_module>::<manager_module>::<@name.classify>
+      #   returns - ::ManageIQ::Providers::Amazon::CloudManager::Vm
+      #
+      # @example derives model_class from @name only
+      #
+      #   @persister_class = ManagerRefresh::Inventory::Persister
+      #   @name = :vms
+      #
+      #   returns ::Vm
+      #
+      # @return [Class | nil] when class doesn't exist, returns nil
+      def auto_model_class
+        model_class = begin
+          # a) Provider specific class
+          provider_module = ManageIQ::Providers::Inflector.provider_module(@persister_class).name
+          manager_module = self.class.name.split('::').last
+
+          class_name = "#{provider_module}::#{manager_module}::#{@name.to_s.classify}"
+          class_name.safe_constantize
+        rescue ::ManageIQ::Providers::Inflector::ObjectNotNamespacedError
+          nil
+        end
+
+        if model_class
+          model_class
+        else
+          # b) general class
+          "::#{@name.to_s.classify}".safe_constantize
+        end
+      end
+
+      # Inventory object attributes are derived from setters
+      #
+      # Can be disabled by options :auto_inventory_attributes => false
+      #   - attributes can be manually set via method add_inventory_attributes()
+      def auto_inventory_attributes
+        return if @properties[:model_class].nil?
+
+        (@properties[:model_class].new.methods - ApplicationRecord.methods).grep(/^[\w]+?\=$/).collect do |setter|
+          setter.to_s[0..setter.length - 2].to_sym
+        end
+      end
+
+      # Evaluates lambda blocks in @builder_params
+      def evaluate_builder_params_lambdas!(persister)
+        if @builder_params
+          @builder_params = @builder_params.transform_values do |value|
+            if value.respond_to?(:call)
+              value.call(persister)
+            else
+              value
+            end
+          end
+        end
+      end
+
+      def network_manager_collections?
+        self.class.network_manager_collections?
+      end
```

Comment:
Is this used anymore?  I don't see it being called and hopefully it isn't needed since this looks too specific for the core class.

Reply:
Can you clean this up in a follow-up?

Reply:
You're right, I tried to use class method and then forget to delete both, they're not used now